kivy/ext/mpl/backend_kivyagg.py

```diff
@@ -0,0 +1,216 @@
+"""
+This is a fully functional do nothing backend to provide a template to
+backend writers. It is fully functional in that you can select it as
+a backend with
+
+  import matplotlib
+  matplotlib.use('Template')
+
+and your matplotlib scripts will (should!) run without error, though
+no output is produced. This provides a nice starting point for
+backend writers because you can selectively implement methods
+(draw_rectangle, draw_lines, etc...) and slowly see your figure come
+to life w/o having to have a full blown implementation before getting
+any results.
+
+Copy this to backend_xxx.py and replace all instances of 'template'
+with 'xxx'. Then implement the class methods and functions below, and
+add 'xxx' to the switchyard in matplotlib/backends/__init__.py and
+'xxx' to the backends list in the validate_backend methon in
+matplotlib/__init__.py and you're off. You can use your backend with::
+
+  import matplotlib
+  matplotlib.use('xxx')
+  from pylab import *
+  plot([1,2,3])
+  show()
+
+matplotlib also supports external backends, so you can place you can
+use any module in your PYTHONPATH with the syntax::
+
+  import matplotlib
+  matplotlib.use('module://my_backend')
+
+where my_backend.py is your module name. This syntax is also
+recognized in the rc file and in the -d argument in pylab, e.g.,::
+
+  python simple_plot.py -dmodule://my_backend
+
+If your backend implements support for saving figures (i.e. has a print_xyz()
+method) you can register it as the default handler for a given file type
+
+  from matplotlib.backend_bases import register_backend
+  register_backend('xyz', 'my_backend', 'XYZ File Format')
+  ...
+  plt.savefig("figure.xyz")
+
+The files that are most relevant to backend_writers are
+
+  matplotlib/backends/backend_your_backend.py
+  matplotlib/backend_bases.py
+  matplotlib/backends/__init__.py
+  matplotlib/__init__.py
+  matplotlib/_pylab_helpers.py
+
+Naming Conventions
+
+  * classes Upper or MixedUpperCase
+
+  * varables lower or lowerUpper
+
+  * functions lower or underscore_separated
+
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+
+import matplotlib
+from matplotlib._pylab_helpers import Gcf
+from matplotlib.backend_bases import RendererBase, GraphicsContextBase,\
+    FigureManagerBase, FigureCanvasBase
+from matplotlib.figure import Figure
+from matplotlib.transforms import Bbox
+from matplotlib.backends.backend_agg import FigureCanvasAgg
+from kivy.input.motionevent import MotionEvent
+
+try:
+    import kivy
```

Comment:
Only the `import kivy` should be under the try/except, the rest should be after it, and outside it.